javagen/src/main/java/com/azure/autorest/model/javamodel/JavaPackage.java

```diff
@@ -178,6 +179,24 @@ protected void addJavaFile(JavaFile javaFile) {
         javaFiles.add(javaFile);
     }
 
+    public void addProtocolExamples(ClientMethod method, MethodGroupClient client, String builderName) {
+        if (method.getProxyMethod().getExamples() == null) {
+            return;
+        }
+        method.getProxyMethod().getExamples().forEach((name, example) -> {
+            String filename = CodeNamer.toPascalCase(CodeNamer.removeInvalidCharacters(name));
+            if (!protocolExampleNameSet.contains(filename)) {
+                JavaFile javaFile = javaFileFactory.createSampleFile(settings.getPackage(), filename);
+                ProtocolExample protocolExample = new ProtocolExample(method, client, builderName, filename, example);
+                Templates.getProtocolSampleTemplate().write(protocolExample, javaFile);
+                javaFiles.add(javaFile);
+                protocolExampleNameSet.add(filename);
+            }
+        });
+    }
+
+    private Set<String> protocolExampleNameSet = new HashSet<>();
```

Comment:
Move this to the top of the class before all methods

Reply:
Done